### PR TITLE
fix(transport): Remove support for OpenSSL

### PR DESCRIPTION
**Pull request by jen20**
This pull request removes support for OpenSSL now that the interop tests can run using Rustls, as discussed in Discord.

### CODE REVIEW

File: tonic/src/transport/endpoint.rs [hunk 1]
```diff
@@ -247,35 +242,23 @@ impl fmt::Debug for ClientTlsConfig {
...
-    #[cfg(feature = "openssl")]
-    pub fn with_openssl() -> Self {
-        Self::new(TlsProvider::OpenSsl)
-    }
-
     /// Creates a new `ClientTlsConfig` using Rustls.
-    #[cfg(feature = "rustls")]
     pub fn with_rustls() -> Self {
-        Self::new(TlsProvider::Rustls)
+        Self::new()
```
Comment (LucioFranco) [inline tonic/src/transport/endpoint.rs]: can we combine the method below into this?

File: tonic/src/transport/server.rs [hunk 2]
```diff
@@ -332,48 +328,30 @@ impl fmt::Debug for Server {
...
-    #[cfg(feature = "openssl")]
-    pub fn with_openssl() -> Self {
-        Self::new(TlsProvider::OpenSsl)
-    }
-
     /// Creates a new `ServerTlsConfig` using Rustls.
-    #[cfg(feature = "rustls")]
     pub fn with_rustls() -> Self {
-        Self::new(TlsProvider::Rustls)
+        Self::new()
```
Comment (LucioFranco) [inline tonic/src/transport/server.rs]: same here?

File: tonic/src/transport/service/tls.rs [hunk 3]
```diff
@@ -132,58 +64,40 @@ impl TlsConnector {
...
-                let dns = DNSNameRef::try_from_ascii_str(self.domain.as_str())
-                    .unwrap()
-                    .to_owned();
-
-                let io = RustlsConnector::from(config.clone())
-                    .connect(dns.as_ref(), io)
-                    .await?;
+        let tls_io = {
+            let dns = DNSNameRef::try_from_ascii_str(self.domain.as_str())
+                .unwrap()
```
Comment (LucioFranco) [inline tonic/src/transport/service/tls.rs]: is this unwrap safe btw?

Comment (jen20) [reply tonic/src/transport/service/tls.rs]: No, pretty sure it's not.

**Comment by jen20** [comment]
@LucioFranco This changes the required checks since interop can now be done as a standard matrix build. Could you update in the repo settings?

**Comment by LucioFranco** [comment]
@jen20 I also removed all the required checks for now.

**Pull request merged**